### ISSUE TITLE
ROOT relies on certain environment variables, breaking distributed usage on certain platforms

### ISSUE THREAD

**Issue by vepadulano** [root-project/root#13811]
### Check duplicate issues.

- [ ] Checked for duplicates

### Description

# TL;DR

ROOT doesn't work on SWAN within any Spark job, unless these environment variables are explicitly set in the worker env beforehand:

* `PATH`:  `/cvmfs/sft.cern.ch/lcg/releases/gcc/11.3.0-ad0f5/x86_64-centos7/bin:/cvmfs/sft.cern.ch/lcg/releases/binutils/2.37-355ed/x86_64-centos7/bin:/usr/bin:/usr/local/bin`
* `ROOT_INCLUDE_PATH`: `/cvmfs/sft-nightlies.cern.ch/lcg/views/devswan/Thu/x86_64-centos7-gcc11-opt/include`

This behaviour has appeared since LCG102, which means ROOT 6.26. On LCG101 (ROOT 6.24.06), the behaviour is not shown.

# The long story

ROOT doesn't work on certain platforms, given the following situation particular situation (which could then be extrapolated to more general ones).

On SWAN, when spawning Spark executors, their environment is cherry-picked to ensure their proper functioning. In particular, only these environment variables are defined:
```python
['CLASSPATH',
 'CONTAINER_ID',
 'HADOOP_COMMON_HOME',
 'HADOOP_CONF_DIR',
 'HADOOP_HDFS_HOME',
 'HADOOP_HOME',
 'HADOOP_TOKEN_FILE_LOCATION',
 'HADOOP_YARN_HOME',
 'HOME',
 'JAVA_HOME',
 'JVM_PID',
 'LANG',
 'LD_LIBRARY_PATH',
 'LOCAL_DIRS',
 'LOCAL_USER_DIRS',
 'LOGNAME',
 'LOG_DIRS',
 'MALLOC_ARENA_MAX',
 'NM_AUX_SERVICE_mapreduce_shuffle',
 'NM_AUX_SERVICE_spark2_shuffle',
 'NM_AUX_SERVICE_spark_shuffle',
 'NM_HOST',
 'NM_HTTP_PORT',
 'NM_PORT',
 'OMP_NUM_THREADS',
 'PATH',
 'PRELAUNCH_ERR',
 'PRELAUNCH_OUT',
 'PWD',
 'PYTHONHASHSEED',
 'PYTHONPATH',
 'PYTHONUNBUFFERED',
 'PYTHON_WORKER_FACTORY_SECRET',
 'SHLVL',
 'SPARK_AUTH_SOCKET_TIMEOUT',
 'SPARK_BUFFER_SIZE',
 'SPARK_DIST_CLASSPATH',
 'SPARK_JOB_ARTIFACT_UUID',
 'SPARK_LOCAL_DIRS',
 'SPARK_PREFER_IPV6',
 'SPARK_REUSE_WORKER',
 'SPARK_USER',
 'SPARK_YARN_STAGING_DIR',
 'USER',
 '_']
```

And in particular these are the contents of  `PATH` inside the worker environment:
![image](https://github.com/root-project/root/assets/15638895/03cc533c-39a9-41f1-b923-1d8dd1ca8000)

A [recent patch](https://github.com/root-project/root/pull/13803) has fixed the `import ROOT` issue:
![image](https://github.com/root-project/root/assets/15638895/cf8df2f6-3d57-44cc-802e-1b9f65d2f9ac)

But anything more than that will fail with:

![image](https://github.com/root-project/root/assets/15638895/f48698a7-6b29-4f0c-9540-6ee292e8b164)

This first issue seems to be fixed by explicitly setting the `ROOT_INCLUDE_PATH` variable to the correct LCG stack include directory on the Spark workers, via the SWAN Spark UI
![image](https://github.com/root-project/root/assets/15638895/b9526d8c-68b1-426c-acf4-665ffa92ec44)

And this is the result:

![image](https://github.com/root-project/root/assets/15638895/9379d5ad-bd53-4867-9654-6106b6e78ed4)

Now a new failure appears with a just slightly more complicated example, which simply opens a TFile with an EOS path
![image](https://github.com/root-project/root/assets/15638895/fd4fb9c3-c20d-4b0e-9a98-7327c70c6dae)

If I add the gcc and binutils directories to `PATH` the file can be opened, specifically I need to set

```
spark.executorEnv.PATH
/cvmfs/sft.cern.ch/lcg/releases/gcc/11.3.0-ad0f5/x86_64-centos7/bin:/cvmfs/sft.cern.ch/lcg/releases/binutils/2.37-355ed/x86_64-centos7/bin:/usr/bin:/usr/local/bin
```

![image](https://github.com/root-project/root/assets/15638895/b49ead64-6044-4e3a-89d5-18e4909dfcd2)

### Reproducer

See the description above

### ROOT version

ROOT master https://github.com/root-project/root/commit/1ed5b2319a500de0d0e5815c2a17615c134c863d, taken from the LCG nightlies

### Installation method

LCG stack

### Operating system

Linux

**Comment by vepadulano** [comment]
The situation on SWAN has improved. Examples using the Spark K8s cluster today 

<img width="820" alt="Image" src="https://github.com/user-attachments/assets/74aa181d-8491-4a03-8bf5-300589e1153f" />

<img width="869" alt="Image" src="https://github.com/user-attachments/assets/a8358ca0-e732-440c-8a70-1c182a0b0373" />

The appropriate environment variables needed for ROOT as well as many other packages of the scientific stack provided via the LCG are now correctly propagated to the Spark workers, as can be seen via the Spark WebUI

<img width="1485" alt="Image" src="https://github.com/user-attachments/assets/5172a835-10cc-4c86-9066-086f67968165" />

This issue can be considered closed thanks to the work of the SWAN team.
